### PR TITLE
Adding push support

### PR DESCRIPTION
**Pull request by stjohnjohnson**
This will trigger the main job on push to the branch.

### CODE REVIEW

**Comment by petey** [comment]
👍  Seems like we could refactor pipeline.get() => sync().

File: plugins/webhooks/github.js [hunk 1]
```diff
@@ -226,6 +226,60 @@ function pullRequestEvent(request, reply) {
...
+
+    // Fetch the pipeline associated with this hook
+    return pipelineFactory.get({ scmUrl })
+        .then(pipeline => {
+            if (!pipeline) {
+                throw boom.notFound('Pipeline does not exist');
+            }
+
+            // sync the pipeline to get the latest jobs
+            return pipeline.sync()
```
Comment (d2lam) [inline plugins/webhooks/github.js]: Doesn't this already create the main job?

Comment (stjohnjohnson) [inline plugins/webhooks/github.js]: What do you mean?

Comment (d2lam) [inline plugins/webhooks/github.js]: I mean pipeline.sync already creates the main job. Why do we do buildFactory.create main again?

Comment (d2lam) [inline plugins/webhooks/github.js]: Line 267

Comment (d2lam) [inline plugins/webhooks/github.js]: Nvm i got confused. You are trying to start the build here.

File: plugins/webhooks/github.js [hunk 2]
```diff
@@ -226,6 +226,60 @@ function pullRequestEvent(request, reply) {
...
+    const jobFactory = request.server.app.jobFactory;
+    const buildFactory = request.server.app.buildFactory;
+    const eventId = request.headers['x-github-delivery'];
+    const payload = request.payload;
+    const repository = hoek.reach(payload, 'repository.ssh_url');
+    const branch = hoek.reach(payload, 'ref').replace(/^refs\/heads\//, '');
+    const sha = hoek.reach(payload, 'after');
+    const username = hoek.reach(payload, 'sender.login');
+    const scmUrl = `${repository}#${branch}`;
+    const apiUri = request.server.info.uri;
```
Comment (nkatzman) [inline plugins/webhooks/github.js]: should this also look for the overridden config value we set?

File: plugins/webhooks/github.js [hunk 3]
```diff
@@ -226,6 +226,60 @@ function pullRequestEvent(request, reply) {
...
+                    return buildFactory.create({ jobId, sha, username, apiUri, tokenGen })
+                        // log build created
+                        .then(build => {
+                            request.log(['webhook-github', eventId, jobId, build.id],
+                                `${name} started ${build.number}`);
+
+                            return reply().code(201);
+                        })
+                        // oops. something went wrong
+                        .catch(err => reply(boom.wrap(err)));
```
Comment (nkatzman) [inline plugins/webhooks/github.js]: do we need this catch if we have it on 279?

File: plugins/webhooks/github.js (hunk 2 above)
Comment (stjohnjohnson) [inline plugins/webhooks/github.js]: nope, it's gone.

File: plugins/webhooks/github.js (hunk 3 above)
Comment (stjohnjohnson) [inline plugins/webhooks/github.js]: Good catch

**Comment by nkatzman** [comment]
👍

**Comment by petey** [comment]
👍

**Pull request merged**